sharding/nodesSetup.go

```diff
@@ -320,41 +327,45 @@ func (ns *NodesSetup) InitialNodesInfoForShard(shardId uint32) ([]nodesCoordinat
 
 // NumberOfShards returns the calculated number of shards
 func (ns *NodesSetup) NumberOfShards() uint32 {
-	return ns.nrOfShards
+	return ns.numberOfShards
 }
 
 // MinNumberOfNodes returns the minimum number of nodes
 func (ns *NodesSetup) MinNumberOfNodes() uint32 {
-	return ns.nrOfShards*ns.MinNodesPerShard + ns.MetaChainMinNodes
+	return ns.computeMinNumberOfNodes()
+}
+
+func (ns *NodesSetup) computeMinNumberOfNodes() uint32 {
+	return ns.numberOfShards*ns.genesisChainParameters.ShardMinNumNodes + ns.genesisChainParameters.MetachainMinNumNodes
 }
 
 // MinShardHysteresisNodes returns the minimum number of hysteresis nodes per shard
 func (ns *NodesSetup) MinShardHysteresisNodes() uint32 {
-	return uint32(float32(ns.MinNodesPerShard) * ns.Hysteresis)
+	return uint32(float32(ns.genesisChainParameters.ShardMinNumNodes) * ns.Hysteresis)
 }
 
 // MinMetaHysteresisNodes returns the minimum number of hysteresis nodes in metachain
 func (ns *NodesSetup) MinMetaHysteresisNodes() uint32 {
-	return uint32(float32(ns.MetaChainMinNodes) * ns.Hysteresis)
+	return uint32(float32(ns.genesisChainParameters.MetachainMinNumNodes) * ns.Hysteresis)
 }
 
 // MinNumberOfNodesWithHysteresis returns the minimum number of nodes with hysteresis
 func (ns *NodesSetup) MinNumberOfNodesWithHysteresis() uint32 {
-	hystNodesMeta := ns.MinMetaHysteresisNodes()
-	hystNodesShard := ns.MinShardHysteresisNodes()
-	minNumberOfNodes := ns.MinNumberOfNodes()
+	hystNodesMeta := ns.genesisChainParameters.MetachainMinNumNodes
```

Comment:
please compute the MinMetaHysteresisNodes
same for hystNodesShard use the MinShardHysteresisNodes

Reply:
fixed :pray: